### ISSUE TITLE
Booting from Hard Disk or FATAL: No bootable medium found! System halted Message On Reboot With VMs

### ISSUE THREAD

**Issue by intelligentgaming** [archlinux/archinstall#185]
Hi there, so I'm testing this out in a VM before I commit it to bare metal, but for some reason I cannot get it to work.

The installation completes successfully, but on reboot it gets stuck on Booting from Hard Disk or FATAL: No bootable medium found! System halted if I specifically force the VM to boot from the virtual hard drive. 

Tried it with both VirtualBox and QEMU/KVM.

I'm hoping it is something daft I am doing, or is this an inherent limitation of running Arch in a VM, as I've probably installed at least a dozen other distributions and never seen this happen. 

Thanks

Ryan

**Comment by dylanmtaylor** [comment]
You need to use an EFI virtual machine. ArchInstall uses systemd-boot as a bootloader which requires UEFI firmware. You can switch this fairly easily in VirtualBox.

**Comment by Torxed** [comment]
I will add a warning if people try to start archinstall without EFI, can't believe I forgot this tbh.
The next version however *(`v2.2.0`)* will support non-EFI platforms as well.

If EFI wasn't the issue, I'd be happy to re-open it.
To get EFI in Qemu/KVM I suggest something like for testing:

```
qemu ... \
-drive if=pflash,format=raw,readonly,file=/usr/share/ovmf/x64/OVMF_CODE.fd \
-drive if=pflash,format=raw,readonly,file=/usr/share/ovmf/x64/OVMF_VARS.fd \
```

**Comment by dylanmtaylor** [comment]
> I will add a warning if people try to start archinstall without EFI, can't believe I forgot this tbh.
> The next version however _(`v2.2.0`)_ will support non-EFI platforms as well.
> 
> If EFI wasn't the issue, I'd be happy to re-open it.
> To get EFI in Qemu/KVM I suggest something like for testing:
> 
> ```
> qemu ... \
> -drive if=pflash,format=raw,readonly,file=/usr/share/ovmf/x64/OVMF_CODE.fd \
> -drive if=pflash,format=raw,readonly,file=/usr/share/ovmf/x64/OVMF_VARS.fd \
> ```
> 
> The fix was merged in with PR #113.

I'm hoping this and the gnome/kde changes can be backported from v2.2.0 to master if they're not going to be shipped with the next ISO as there are huge quality of life changes in that branch.

**Comment by intelligentgaming** [comment]
Thank you for your help. 

All working now.
